app/controllers/api/v1/api_controller.rb

```diff
@@ -4,14 +4,23 @@ class ApiController < ActionController::API
       before_action :auth
 
       def current_user
-        @current_user ||= User.find_by(token: token)
+        @current_user ||= if doorkeeper_token
+                            User.find(doorkeeper_token.resource_owner_id)
+                          else
+                            User.find_by(token: token)
+                          end
       end
 
       private
 
       def auth
-        return render json: { message: 'Missing Token' }, status: :unauthorized if token.nil?
-        return render json: { message: 'Invalid Token' }, status: :unauthorized unless User.exists?(token: token)
+        return render json: { message: 'Missing Token' }, status: :unauthorized unless token || doorkeeper_token
+
+        unless valid_doorkeeper_token? || (token && User.exists?(token: token))
```

Comment:
For `token` we are also checking if the user exists, don't we need to do the same thing when it's a doorkeeper token?

Reply:
This will not be necessary https://github.com/Codeminer42/Punchclock/pull/116/commits/68d448b0fce98255f0246e7b10f3e3eb1ec7d5c7